packages/calcite-components/src/components/input-number/input-number.tsx

```diff
@@ -675,6 +675,11 @@ export class InputNumber
       return;
     }
 
+    // identifies French AZERTY keyboard input
```

Comment:
Can you generalize this comment? I think this would apply to other keyboard layouts. Bonus points if you use a self-documenting const for this. ⭐

Actually, do we need to care about the Shift key being pressed? Wouldn't we want to process any number from `KeyboardEvent.key`? I tracked that part of logic down to https://github.com/Esri/calcite-components/issues/2127. @eriklharper Do you recall why we bail when the Shift key is pressed? This could possibly extend to the Alt, Ctrl and Meta keys.

Reply:
The reason for verifying Shift key here is to make sure the numbers are being returned. In AZERTY keyboard , users need to keep shift key activated for numbers to be printed.  

Ideally we would like to process any number but as you mentioned there is a custom logic that prevents numberKeys, ArrowKeys and Meta keys with Tab+Shift which results in the bug #6854.  I wonder if there will be a case where the `event.key` will return  a numberKey when Shift+Tab or Shift key is activated. 

Either we need to remove `numberKeys` from supportedKeys array or remove the check for `shiftKey`. With input and input-number , i am very hesitant to remove any existing code since it is very impractical to iterate tests for every single locale we support.

Reply:
> The reason for verifying Shift key here is to make sure the numbers are being returned. In AZERTY keyboard , users need to keep shift key activated for numbers to be printed.

>  I wonder if there will be a case where the event.key will return a numberKey when Shift+Tab or Shift key is activated.

This is what I'm saying. In AZERTY, when the key combination for a number is pressed, you get a number in `KeyboardEvent.key`. 

> or remove the check for shiftKey. 

Unless I'm missing something, I think this is the way to go (`Shift` + `Tab` being the exception).